src/main/scala/tech/cryptonomic/conseil/tezos/TezosDatabaseOperations.scala

```diff
@@ -292,4 +294,47 @@ object TezosDatabaseOperations extends LazyLogging {
   def selectDistinctLike(table: String, column: String, matchingString: String)(implicit ec: ExecutionContext): DBIO[List[String]] = {
     sql"""SELECT DISTINCT #$column FROM #$table WHERE #$column LIKE '%#$matchingString%'""".as[String].map(_.toList)
   }
+
+
+  /** Implicit value that allows getting table row as Map[String, Any] */
+  private implicit val getMap: GetResult[Map[String, Any]] = GetResult[Map[String, Any]](positionedResult => {
+    val metadata = positionedResult.rs.getMetaData
+    (1 to positionedResult.numColumns).flatMap(i => {
+      val columnName = metadata.getColumnName(i).toLowerCase
+      val columnValue = positionedResult.nextObjectOption
+      columnValue.map(columnName -> _)
+    }).toMap
+  })
+
+  /**
+    * Selects elements filtered by the predicates
+    * THIS METHOD IS VULNERABLE TO SQL INJECTION
+    * @param table          name of the table
+    * @param columns        list of column names
+    * @param predicates     list of predicates for query to be filtered with
+    * @return               list of map of [string, any], which represents list of rows as a map of column name to value
+    */
+  def selectWithPredicates(table: String, columns: List[String], predicates: List[Predicate])(implicit ec: ExecutionContext):
+  DBIO[List[Map[String, Any]]] = {
+    import tech.cryptonomic.conseil.util.DatabaseUtil._
+    val pred = predicates.map { p =>
+      concat(
+        sql""" AND #${p.field} #${mapOperationToSQL(p.operation, p.inverse)} """,
+        List(values(p.set.map(_.toString)))
+      )
+    }
+    val query = sql"""SELECT #${columns.mkString(",")} FROM #$table WHERE true """
```

Comment:
Can you refactor the SQL generation into separate functions and write unit tests for them?

Reply:
I've extracted two methods, but I could test only one of them, I don't know how to make viable tests for conditions - I could only test basic query. The main method which prepares the query `selectWithPredicates` has tests.